docs/solvers/nn.rst

```diff
@@ -10,12 +10,23 @@ Neural Dual
 .. autosummary::
     :toctree: _autosummary
 
-    neuraldual.NeuralDualSolver
+    neuraldual.W2NeuralDual
 
-ICNN
-----
+Models
+------
 .. autosummary::
     :toctree: _autosummary
 
-    icnn.ICNN
-    layers.PositiveDense
+    models.NeuralTrainState
```

Comment:
I'd exclude `NeuralTrainState` from the docs, as it is too technical and doesn't provide much info (given it's well documented in `W2NeuralDual`).